executable-spec/src/Cardano/Ledger/Generators.hs

```diff
@@ -4,19 +4,34 @@ module Cardano.Ledger.Generators
   ( kGen
   , currentSlotGen
   , participantsGen
+  , voteTGen
+  , stakeDistGen
   )
 where
 
 import           Control.Arrow ((&&&))
 import           Data.Bimap (Bimap)
 import qualified Data.Bimap as Bimap
+import           Data.Map (Map)
+import qualified Data.Map.Strict as Map
 import           Hedgehog (Gen)
 import qualified Hedgehog.Gen as Gen
 import qualified Hedgehog.Range as Range
 
 import qualified Ledger.Core as Core
+import           Cardano.Ledger.Spec.STS.Update.Data (VThreshold, Stake)
 
 
+stakeDistGen :: Gen (Map Core.VKey Stake)
+stakeDistGen = do
+  p <- participantsGen
+  let vkeys = Bimap.keys p
+  stks <- Gen.list (Range.singleton $ length vkeys) (Gen.word64 (Range.linear 1 1000))
+  pure $ Map.fromList $ zip vkeys stks
+
+voteTGen :: Gen VThreshold
+voteTGen = Gen.integral (Range.constant 50 75)
```

Comment:
We should probably use the definition of `tauV` in terms of adversarial stake `P`.

Reply:
In fact, we only need a single parameter, the actual ratio of adversary stake `r_a = T/(T+h_a)`
because `\tau_v = 50 * (1 + r_a)`
In that case, we need to generate values for `r_a` in the range of `[0, 0.49]`